modules/stochastic_tools/src/samplers/BayesianActiveLearningSampler.C

```diff
@@ -0,0 +1,92 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BayesianActiveLearningSampler.h"
+#include "Normal.h"
+#include "Uniform.h"
+
+registerMooseObject("StochasticToolsApp", BayesianActiveLearningSampler);
+
+InputParameters
+BayesianActiveLearningSampler::validParams()
+{
+  InputParameters params = PMCMCBase::validParams();
+  params.addClassDescription("Fast Bayesian inference with the parallel active learning (partly "
+                             "inspired from El Gammal et al. 2023).");
+  params.addRequiredParam<ReporterName>(
+      "sorted_indices", "The sorted sample indices in order of importance to evaluate the subApp.");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "num_tries",
+      "num_tries>0",
+      "Number of samples to propose in each iteration (not all are sent for subApp evals).");
+  return params;
+}
+
+BayesianActiveLearningSampler::BayesianActiveLearningSampler(const InputParameters & parameters)
+  : PMCMCBase(parameters),
+    _sorted_indices(getReporterValue<std::vector<unsigned int>>("sorted_indices")),
+    _num_tries(getParam<unsigned int>("num_tries"))
+{
+  _inputs_test.resize(_num_tries, std::vector<Real>(_priors.size()));
+  _var_test.resize(_num_tries);
+  _sample_vector.resize(_priors.size());
+}
+
+void
+BayesianActiveLearningSampler::fillVector(std::vector<Real> & vector,
+                                          const unsigned int & seed_value)
+{
+  for (unsigned int i = 0; i < _priors.size(); ++i)
+    vector[i] = _priors[i]->quantile(getRand(seed_value));
+}
+
+const std::vector<std::vector<Real>> &
+BayesianActiveLearningSampler::getSampleTries() const
+{
+  return _inputs_test;
+}
+
+const std::vector<Real> &
+BayesianActiveLearningSampler::getVarSampleTries() const
+{
+  return _var_test;
+}
+
+void
+BayesianActiveLearningSampler::proposeSamples(const unsigned int seed_value)
+{
+  /* If step is 1, randomly generate the samples.
+  Else, generate the samples informed by the GP from the reporter "sorted_indices" */
+  for (dof_id_type i = 0; i < _num_parallel_proposals; ++i)
+  {
+    if (_t_step <= 1)
+    {
+      fillVector(_sample_vector, seed_value);
+      _new_samples[i] = _sample_vector;
```

Comment:
I don't see a reason why `_sample_vector` is needed. Can we not just do:
```suggestion
      fillVector(_new_samples[i], seed_value);
```